test/models/ocelot/stanzas/ChoiceStanzaSpec.scala

```diff
@@ -0,0 +1,229 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.ocelot.stanzas
+
+import base.BaseSpec
+import play.api.libs.json._
+import models.ocelot._
+import models.ocelot.errors.{GuidanceError, UnknownTestType}
+
+class ChoiceStanzaSpec extends BaseSpec {
+
+  val stanzaType = "ChoiceStanza"
+  val valueType = "scalar"
+  val pageNameLabel = "PageName"
+  val pageName = "Telling HMRC about extra income"
+  val pageUrlLabel = "PageUrl"
+  val pageUrl = "/rent/less-than-1000/do-you-want-to-use-the-rent-a-room-scheme"
+  val next = Seq("40", "41")
+  val stack = "false"
+
+  val onePageJsonWithInvalidTestType: JsValue = Json.parse(
+    """
+      |{
+      |  "meta": {
+      |    "title": "Customer wants to make a cup of tea",
+      |    "id": "oct90001",
+      |    "ocelot": 1,
+      |    "lastAuthor": "000000",
+      |    "lastUpdate": 1500298931016,
+      |    "version": 4,
+      |    "filename": "oct90001.js",
+      |    "titlePhrase": 8
+      |  },
+      |  "howto": [],
+      |  "contacts": [],
+      |  "links": [],
+      |  "flow": {
+      |    "start": {
+      |      "type": "PageStanza",
+      |      "url": "/feeling-bad",
+      |      "next": ["3"],
+      |      "stack": true
+      |    },
+      |    "3": {
+      |      "type": "ChoiceStanza",
+      |      "tests": [
+      |        {
+      |          "left": "VAL-1",
+      |          "test": "UnknownType",
+      |          "right": "VAL-2"
+      |        }],
+      |      "next": [
+      |        "2", "3"
+      |      ],
+      |      "stack": true
+      |    },
+      |    "2": {
+      |      "type": "InstructionStanza",
+      |      "text": 0,
+      |      "next": [
+      |        "end"
+      |      ],
+      |      "stack": true
+      |    },
+      |    "end": {
+      |      "type": "EndStanza"
+      |    }
+      |  },
+      |  "phrases": [
+      |    ["Ask the customer if they have a tea bag", "Welsh, Ask the customer if they have a tea bag"],
+      |    ["Do you have a tea bag?", "Welsh, Do you have a tea bag?"],
+      |    ["Yes - they do have a tea bag", "Welsh, Yes - they do have a tea bag"],
+      |    ["No - they do not have a tea bag", "Welsh, No - they do not have a tea bag"],
+      |    ["Ask the customer if they have a cup", "Welsh, Ask the customer if they have a cup"],
+      |    ["Do you have a cup?", "Welsh, Do you have a cup?"],
+      |    ["yes - they do have a cup ", "Welsh, yes - they do have a cup "],
+      |    ["no - they don’t have a cup", "Welsh, no - they don’t have a cup"],
+      |    ["Customer wants to make a cup of tea", "Welsh, Customer wants to make a cup of tea"]
+      |  ]
+      |}
+    """.stripMargin
+  )
+
+
+  val validChoiceStanzaJson: JsObject = Json
```

Comment:
Strictly speaking I think next should have three ids. These match to
1) test1 is true
2) test2 is true
3) neither test is true

Reply:
Yes, updated